tests/php/src/Admin/SiteHealthTest.php

```diff
@@ -589,6 +652,174 @@ public function test_add_extensions() {
+						'color' => 'orange',
+					],
+					'test'   => 'amp_page_cache',
+					'status' => 'recommended',
+					'label'  => 'Page caching is not detected',
+				],
+			],
+			'server-cache'          => [
+				'request_headers' => [
+					'x-amp-random-number' => '1345',
+				],
+				'expected'        => [
+					'badge'  => [
+						'label' => 'AMP',
+						'color' => 'orange',
+					],
+					'test'   => 'amp_page_cache',
+					'status' => 'recommended',
+					'label'  => 'Page caching is detected but client caching headers are missing',
+				],
+			],
+			'server-cache-with-age' => [
+				'request_headers' => [
+					'age' => '1345',
+				],
+				'expected'        => [
+					'badge'  => [
+						'label' => 'AMP',
+						'color' => 'orange',
+					],
+					'test'   => 'amp_page_cache',
+					'status' => 'recommended',
+					'label'  => 'Page caching is detected but client caching headers are missing',
+				],
+			],
+			'full-cache'            => [
+				'request_headers' => [
+					'x-amp-random-number' => '1345',
+					'expires'             => 'Wed, 11 Jan 1984 12:00:00 GMT',
+				],
+				'expected'        => [
+					'badge'  => [
+						'label' => 'AMP',
+						'color' => 'green',
+					],
+					'test'   => 'amp_page_cache',
+					'status' => 'good',
+					'label'  => 'Page caching is detected',
+				],
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider get_page_cache_data
+	 * @covers ::page_cache()
+	 */
+	public function test_page_cache( $request_headers, $expected ) {
+
+		/*
+		 * Mock the http request.
+		 */
+		$callback_wp_remote = static function () use ( $request_headers ) {
+
+			return [
+				'headers' => $request_headers,
+			];
+		};
+		add_filter( 'pre_http_request', $callback_wp_remote );
+
+		$actual = $this->instance->page_cache();
+		$this->assertArrayHasKey( 'description', $actual );
+		$this->assertArrayHasKey( 'actions', $actual );
+		unset( $actual['description'] );
+		unset( $actual['actions'] );
+
+		$this->assertEquals(
+			$expected,
+			$actual
+		);
+
+		remove_filter( 'pre_http_request', $callback_wp_remote );
+	}
+
+	/**
+	 * Data provider for $this->test_get_page_cache_status()
+	 *
+	 * @return array[]
+	 */
+	public function get_data_to_test_get_page_cache_status() {
+
+		return [
+			'no-cache'              => [
+				'request_headers' => [],
+				'expected'        => [
+					'server_caching' => false,
+					'client_caching' => false,
+				],
+			],
+			'server-cache'          => [
+				'request_headers' => [
+					'x-amp-random-number' => '1345',
+				],
+				'expected'        => [
+					'server_caching' => true,
+					'client_caching' => false,
+				],
+			],
+			'server-cache-with-age' => [
+				'request_headers' => [
+					'age' => '1345',
+				],
+				'expected'        => [
+					'server_caching' => true,
+					'client_caching' => false,
+				],
+			],
+			'full-cache'            => [
+				'request_headers' => [
+					'x-amp-random-number' => '1345',
+					'expires'             => 'Wed, 11 Jan 1984 12:00:00 GMT',
```

Comment:
This expiration date is in the past, so it should indicate that no caching I'd expect.